src/governor.cairo

```diff
@@ -447,6 +448,29 @@ pub mod Governor {
 
             replace_class_syscall(class_hash).unwrap();
         }
+
+        fn _migrate_old_config_storage(ref self: ContractState) {
+            let old_config_storage_address = storage_base_address_from_felt252(selector!("config"));
+            let old_config: Config = Store::read(0, old_config_storage_address).unwrap();
+            // voting period of 0 is assumed to be invalid
+            assert(old_config.voting_period.is_non_zero(), 'NO_OLD_CONFIG');
+            self.config_versions.write(0, old_config);
+            self.emit(Reconfigured { version: 0, new_config: old_config });
+            Store::write(
+                0,
+                old_config_storage_address,
+                Config {
+                    voting_start_delay: 0,
+                    voting_period: 0,
+                    voting_weight_smoothing_duration: 0,
+                    quorum: 0,
+                    proposal_creation_threshold: 0,
+                    execution_delay: 0,
+                    execution_window: 0
+                }
+            )
```

Comment:
code looks good! 
can you explain me what you mean exactly by `// Migrates to the fixed storage layout, i.e. the storage layout from before v2.1.0` in the interface?

For me you just removed the `config` variable in storage and store the current config (and potential configs associated with other versions) in `config_versions` mapping

Reply:
v2.1.0 is referring to the governor release, not cairo/scarb

Since we added versioned configs, we have a new storage layout

But having a storage variable for config as well as another variable for versioned configs is less elegant than just having all configs be versioned (and the initial config is version 0)

Reply:
All clear thanks a lot!